### PR TITLE
Fixed bug in payment flow

### PR DESCRIPTION
**Pull request by antonioperic**
I noticed that if you are using CreditCard for payment and if you change your mind on page where you need to enter your credit card, jump back to payment methods and pick something else for example offline that there is bug and you cannot complete payment. This PR fix it

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
@makasim WDYT?

File: src/Sylius/Bundle/CoreBundle/EventListener/OrderPaymentListener.php [hunk 1]
```diff
@@ -71,6 +71,8 @@ public function createOrderPayment(GenericEvent $event)
     {
         if (false === $this->getOrder($event)->getLastPayment()) {
             $this->paymentProcessor->createPayment($this->getOrder($event));
+        } else {
+            $this->getOrder($event)->getLastPayment()->setDetails(array());
         }
```
Comment (makasim) [inline src/Sylius/Bundle/CoreBundle/EventListener/OrderPaymentListener.php]: I do not think we have to change the last payment but rather create a new one so an admin can see the whole history, like user selected one payment but than changed his mind and selected another one

Comment (antonioperic) [inline src/Sylius/Bundle/CoreBundle/EventListener/OrderPaymentListener.php]: then just remove if and always create new?

Comment (makasim) [inline src/Sylius/Bundle/CoreBundle/EventListener/OrderPaymentListener.php]: @antonioperic yeap sounds better

**Comment by pjedrzejewski** [comment]
Thank you Antonio! :+1:

**Pull request merged**